client/src/components/Header/components/nav-links.tsx

```diff
@@ -16,33 +16,39 @@ import {
   faExternalLinkAlt
 } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import React, { Component, Fragment } from 'react';
+import React, { Component, Fragment, createRef, Ref } from 'react';
 import { TFunction, withTranslation } from 'react-i18next';
 import { connect } from 'react-redux';
 import envData from '../../../../../config/env.json';
 import {
   availableLangs,
-  getLangName
+  LangNames,
+  LangCodes
 } from '../../../../../config/i18n/all-langs';
 import { hardGoTo as navigate } from '../../../redux';
 import { updateUserFlag } from '../../../redux/settings';
 import createLanguageRedirect from '../../create-language-redirect';
 import { Link } from '../../helpers';
 import { Themes } from '../../settings/theme';
+import LanguageGlobe from '../../../assets/icons/language-globe';
 
 const { clientLocale, radioLocation, apiLocation } = envData;
 
 const locales = availableLangs.client;
 
 export interface NavLinksProps {
   displayMenu?: boolean;
+  isLanguageMenuDisplayed?: boolean;
   fetchState?: { pending: boolean };
   i18n: Object;
   t: TFunction;
-  toggleDisplayMenu?: React.MouseEventHandler<HTMLButtonElement>;
+  hideMenu: () => void;
   toggleNightMode: (x: any) => any;
   user?: Record<string, unknown>;
   navigate?: (location: string) => void;
+  showLanguageMenu?: (elementToFocus: HTMLAnchorElement) => void;
```

Comment:
I think the argument for many of the functions that operate in the nav-links should be a 'HTMLButtonElement'. 

That being said, there are a lot of TS rules that are being ignored in the navigation related components. I don't mind handling in a whole another pr.

Reply:
You are correct. Not sure why I used `HTMLAnchorElement` when all of the items in the language menu are buttons? I guess I was just thinking of the main nav menu in which most of the items are links.